sdk/tables/test-resources.json

```diff
@@ -95,6 +95,7 @@
           ],
           "defaultAction": "Allow"
         },
+        "allowSharedKeyAccess": false,
```

Comment:
This is not required for the fix, but it is a good change that was discovered during writing this PR.
It is one line, so I am including it in this PR.
Tables do no longer have connection string support, so Azure RM template does not need to support them.
Having this allows Azure SDK devs to deploy Tables test resources from their dev machines without any error messages.